be/src/olap/rowset/segment_writer.cpp

```diff
@@ -227,11 +231,10 @@ OLAPStatus SegmentWriter::finalize(uint32_t* segment_file_size) {
     }
 
     // check disk capacity
-    if (data_dir->reach_capacity_limit((int64_t) file_header.file_length())) {
+    if (data_dir != nullptr && data_dir->reach_capacity_limit((int64_t) file_header.file_length())) {
          return OLAP_ERR_DISK_REACH_CAPACITY_LIMIT;
     }
 
-    data_dir->add_pending_ids(ROWSET_ID_PREFIX + _segment_group->rowset_id().to_string());
```

Comment:
Is it OK to remove this operation?

Reply:
I think it can be done in another pr.

Reply:
Why?
I think you should not introduce new problem in on PR.
When we need do in another PR, it must assure that this would not make our code worse.